docs/source-app/workflows/add_web_ui/panel/app_pull_state.py

```diff
@@ -0,0 +1,66 @@
+"""This app demonstrates how the PanelFrontend can read/ pull the global app
+state and display it"""
+import datetime
+
+import lightning as L
+import panel as pn
+from panel_frontend import PanelFrontend
+from pathlib import Path
+import json
+
+DATETIME_FORMAT = "%Y-%m-%d, %H:%M:%S.%f"
+
+APP_STATE = Path(__file__).parent / "app_state.json"
+
+def read_app_state():
+    with open(APP_STATE) as json_file:
+        return json.load(json_file)
+
+def save_app_state(state):
+    with open(APP_STATE, 'w') as outfile:
+        json.dump(state, outfile)
+
+def to_string(value: datetime.datetime)->str:
+    return value.strftime(DATETIME_FORMAT)
+
+def render_fn(self):
+    global_app_state_pane = pn.pane.JSON(depth=2)
+    last_local_update_pane = pn.pane.Str()
+    
+    def update():
+        last_local_update_pane.object= "last local update: " + to_string(datetime.datetime.now())
+        
+        # Todo: Figure out the right way to read/ pull the app state
+        global_app_state_pane.object = read_app_state()

```

Comment:
@tchaton . The key in this app is for me to figure out how the Panel app can read/ pull the global app state. What is the *right way* to do this.

In this implementation I just save and read as json file on a regular basis.

Reply:
@tchaton points me to a dash example: https://lightning.ai/lightning-docs/workflows/add_web_ui/dash/intermediate.html#interact-with-dash-from-a-component. But as far as I can see it only displays the *local* app `LightningWork` state. But I want to display the global state.